main/tests/server/src/com/google/refine/commands/row/AddRowsCommandTests.java

```diff
@@ -0,0 +1,26 @@
+
+package com.google.refine.commands.row;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.google.refine.commands.CommandTestBase;
+
+public class AddRowsCommandTests extends CommandTestBase {
+
+    @BeforeMethod
+    public void setUpCommand() {
+        command = new AddRowsCommand();
+    }
+
+    @Test
+    public void testCSRFProtection() throws ServletException, IOException {
+        command.doPost(request, response);
+        assertCSRFCheckFailed();
+    }
+
```

Comment:
Would it be possible to add a test which demonstrates / tests the REST API for this command?

That would also help document what the format is.

Reply:
Completely agree, but I'm not sure how to do that with Mockito. I've only done API testing with [REST Assured](https://rest-assured.io/) / Postman. Is this something I'd write a Cypress test for?

Reply:
You can see an example of how to set up a request parameter for Mockito here: 
https://github.com/OpenRefine/OpenRefine/blob/2c7b5007e3a2dc496bf85c10a28cad864cff7651/main/tests/server/src/com/google/refine/commands/row/GetRowsCommandTest.java#L107-L109

You could use Cypress, but it's more straightforward to do it as a unit test.